File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
         
        string r="";
        
        
        for (char c: s){
            if (isalnum(c) && c!= ' ') r+= tolower(c);
        }

        int n= r.length();

        for (int i=0; i< n/2; i++){
            if(r[i]!= r[n-i-1]){
                return false;
            }
           
        }

        return true;
    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """

        st=''
        for i in s:
            if (i.isalnum()):
                st+=i.lower()
            
        n= len (st)
        i=0
        j= n-1

        while(i<=j):
            if (st[i]== st[j]):
                i+=1
                j-=1
            else:
                return False

        return True
        